pkg/apis/core/v1/types.go

```diff
@@ -76,6 +77,29 @@ type (
 		Items           []Function `json:"items"`
 	}
 
+	//FunctionStatus ...
+	FunctionStatus struct {
+		// FnStatus is the function status.
+		FnStatus BuildStatus `json:"functionstatus,omitempty"`
+
+		// EnvStatus is the environment status
+		EnvStatus BuildStatus `json:"envStatus,omitempty"`
+
+		// EnvBuildLog stores build log of the environment
+		EnvBuildLog string `json:"envbuildlog,omitempty"`
```

Comment:
My thinking behind adding `EnvBuildLog` and `PkgBuildLog` was:
1) If function fails; user should get an overview of why it failed
2) So, currently; I am populating these fields only in case of errors
3) Upon your comment; I realised that the name - `EnvBuildLog` - may not be appropriate and can be renamed to something like- `EnvStatusLogs` may be. 
In summary; I think we should keep these log fields. Please let me know your opinion on it.